experimental/packages/opentelemetry-sdk-node/src/start.ts

```diff
@@ -54,13 +84,141 @@ export function startNodeSDK(sdkOptions: SDKOptions): {
 
   const shutdownFn = async () => {
     const promises: Promise<unknown>[] = [];
+    if (loggerProvider) {
+      promises.push(loggerProvider.shutdown());
+    }
     await Promise.all(promises);
   };
   return { shutdown: shutdownFn };
 }
 const NOOP_SDK = {
   shutdown: async () => {},
 };
+
+function setupResource(
+  config: ConfigurationModel,
+  sdkOptions: SDKOptions
+): Resource {
+  let resource: Resource = sdkOptions.resource ?? defaultResource();
+  const autoDetectResources = sdkOptions.autoDetectResources ?? true;
+  let resourceDetectors: ResourceDetector[];
+
+  if (!autoDetectResources) {
+    resourceDetectors = [];
+  } else if (sdkOptions.resourceDetectors != null) {
+    resourceDetectors = sdkOptions.resourceDetectors;
+  } else if (config.node_resource_detectors) {
+    resourceDetectors = getResourceDetectorsFromConfiguration(config);
+  } else {
+    resourceDetectors = [
+      envDetector,
+      processDetector,
+      hostDetector,
+      osDetector,
+      serviceInstanceIdDetector,
+    ];
+  }
+
+  if (autoDetectResources) {
+    const internalConfig: ResourceDetectionConfig = {
+      detectors: resourceDetectors,
+    };
+
+    resource = resource.merge(detectResources(internalConfig));
+  }
+
+  resource =
+    sdkOptions.serviceName === undefined
+      ? resource
+      : resource.merge(
+          resourceFromAttributes({
+            [ATTR_SERVICE_NAME]: sdkOptions.serviceName,
+          })
+        );
+
+  return resource;
+}
+
+function setupLoggerProvider(
+  config: ConfigurationModel,
+  sdkOptions: SDKOptions,
+  resource: Resource | undefined
+): LoggerProvider | undefined {
+  const logProcessors =
+    sdkOptions.logRecordProcessors ??
+    getLogRecordProcessorsFromConfiguration(config);
+
+  if (logProcessors) {
+    const loggerProvider = new LoggerProvider({
+      resource: resource,
+      processors: logProcessors,
+    });
+
+    logs.setGlobalLoggerProvider(loggerProvider);
+    return loggerProvider;
+  }
+  return undefined;
+}
+
+function getLogRecordExporter(
+  exporter: LogRecordExporterModel
+): LogRecordExporter {
+  if (exporter.otlp_http) {
+    if (exporter.otlp_http.encoding === 'json') {
+      return new OTLPHttpLogExporter({
+        compression:
+          exporter.otlp_http.compression === 'gzip'
+            ? CompressionAlgorithm.GZIP
+            : CompressionAlgorithm.NONE,
+      });
+    }
+    if (exporter.otlp_http.encoding === 'protobuf') {
+      return new OTLPProtoLogExporter();
+    }
+    diag.warn(`Unsupported OTLP logs protocol. Using http/protobuf.`);
+    return new OTLPProtoLogExporter();
+  } else if (exporter.otlp_grpc) {
+    return new OTLPGrpcLogExporter();
+  } else if (exporter.console) {
+    return new ConsoleLogRecordExporter();
+  }
+  diag.warn(`Unsupported Exporter value. Using OTLP http/protobuf.`);
+  return new OTLPProtoLogExporter();
+}
+
+function getLogRecordProcessorsFromConfiguration(
```

Comment:
Similar to above, we may want this in utils.

Reply:
done